### PR TITLE
Minimally Viable Quiz SPA and Various Other Updates

### PR DESCRIPTION
**Pull request by elmq0022**
I realize now this is probably too much for on pull.  Anyway, here's what's going on in this pull request.  

1. Finalized an SPA that works pretty well.  
1. Added a get_absolute_url to the dictionary.models.Entry. 
1. Used the get_absolute_url method wherever I could.  
1. Update the EntryView to use an UpdateView and a SuccessMessageMixin
1. Tried to stop tracking all the css/js assets so this looks like a python repo again.
1. Added some information to the README.md file.  
1. Added some more site links.

Thanks!

### CODE REVIEW

File: accounts/views.py [hunk 1]
```diff
@@ -43,13 +43,22 @@ class SignUpView(generic.CreateView):
 class Home(LoginRequiredMixin, TemplateView):
     template_name = 'accounts/home.html'
 
-    def top_characters(self):
+    def get_top_characters(self):
         return Count.objects.filter(count_type=Count.CHARACTER).all().order_by('-count')[:5] 
 
-    def top_words(self):
+    def get_top_words(self):
         return Count.objects.filter(count_type=Count.WORD).all().order_by('-count')[:5]
```
Comment (jordanbettis) [inline accounts/views.py:50]: One thing I like to do when I pull data that won't change like this, is to cache it in memory by turning it into a property excessor that is evaluated lazily:

```python
@property
def top_words(self):
    if not hasattr(self, "_top_words"):
        self._top_words = Count.objects.filter(count_type=Count.WORD).all().order_by('-count')[:5]
    return self._top_words
```
Then down below you can just refer to it in each case by `self.top_words` and not have to worry about hitting the database.

File: hanyu/templatetags/AdminLTE.py [hunk 2]
```diff
@@ -8,24 +8,41 @@
...
-
     context['heading'] = heading
-    context['links'] = links
+    return context
+
+
+@register.inclusion_tag('_menu_tree_item.html', takes_context=True)
+def menu_tree_item(context, url_namespace, description, *args):
+    '''
+    description    := list of sub-menu link descriptions
```
Comment (jordanbettis) [inline hanyu/templatetags/AdminLTE.py:26]: I'm not sure where this docstring syntax comes from, but if you want to use sphynx you should use the ReStructured Text format:

http://thomas-cokelaer.info/tutorials/sphinx/docstring_python.html

File: quiz/views.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        uid = data['uid']
+        quiz = Quiz.objects.get(uid=uid)
+        answers = quiz.answer_set.all()
+        correct_answer = answers.filter(correct=True).all()[0]
+        quiz.response = Quiz.CORRECT if correct_answer.pk == data['answer_pk'] else Quiz.INCORRECT
+        quiz.save()
+        result = {'correct_pk': correct_answer.pk}
+        return HttpResponse(json.dumps(result), content_type="application/json")
+
+    if request.method == "GET":
```
Comment (jordanbettis) [inline quiz/views.py]: You could probably just make this an "else"

**Review by jordanbettis — APPROVED** [review]
Looking good. I like the unit tests

**Comment by elmq0022** [comment]
Good points as always.  Thanks!

**Pull request merged**